src/_gettsim_tests/test_jax.py

```diff
@@ -0,0 +1,518 @@
+    (f5, f5_exp, (x,)),
+    (f6, f6_exp, (flag, another_flag)),
+    (f7, f7_exp, (x,)),
+    (f8, f8_exp, (x,)),
+    (f9, f9, (x,)),
+    (f10, f10_exp, (x,)),
+    (f11, f11_exp, (x,)),
+    (f12, f12_exp, (x,)),
+    (f13, f13_exp, (x,)),
+    (f14, f14_exp, (x,)),
+    (f15, f15_exp, (x,)),
+    (f16, f16_exp, (x,)),
+    (f17, f17_exp, (x,)),
+    (f18, f18_exp, (x,)),
+]
+
+
+# ======================================================================================
+# Tests (no error)
+# ======================================================================================
+
+
+@pytest.mark.parametrize("func, expected, args", TEST_CASES)
+def test_change_if_to_where_source(func, expected, args):  # noqa: ARG001
+    exp = inspect.getsource(expected)
+    exp = exp.replace("_exp", "")
+    got = make_vectorizable_source(func, backend="numpy")
+    assert string_equal(exp, got)
+
+
+@pytest.mark.parametrize("func, expected, args", TEST_CASES)
+def test_change_if_to_where_wrapper(func, expected, args):
+    got_func = make_vectorizable(func, backend="numpy")
+    got = got_func(*args)
+    exp = expected(*args)
+    assert_array_equal(got, exp)
+
+
+# ======================================================================================
+# Test correct error raising
+# ======================================================================================
+
+
+def g1(x):
+    # function with multiple operations in the if-clause
+    a = 0
+    b = 1
+    if x < 0:
+        a = 1
+        b = 0
+    return a + b
+
+
+def g2(x):
+    # function with illegal operations in the if-clause
+    if x < 0:
+        print(x)  # noqa: T201
+    else:
+        print(not x)  # noqa: T201
+
+
+def g3(x):
+    # return statement in if-body but no else clause
+    if x < 0:
+        return 0
+    return 1
+
+
+def g4(x):
+    # max with three arguments
+    return max(x, 0, 1)
+
+
+def test_notimplemented_error():
+    with pytest.raises(NotImplementedError):
+        make_vectorizable(f1, backend="dask")
+
+
+@pytest.mark.parametrize("func", [g1, g2, g3, g4])
+def test_unallowed_operation_source(func):
+    with pytest.raises(TranslateToVectorizableError):
+        make_vectorizable_source(func, backend="numpy")
+
+
+@pytest.mark.parametrize("func", [g1, g2, g3, g4])
+def test_unallowed_operation_wrapper(func):
+    with pytest.raises(TranslateToVectorizableError):
+        make_vectorizable(func, backend="numpy")
+
+
+# ======================================================================================
+# Test that functions defined in gettsim can be made vectorizable
+# ======================================================================================
+
+
+gettsim_functions = _load_functions(sources=PATHS_TO_INTERNAL_FUNCTIONS)
+
+not_convertible_yet = {
+    "wohngeld_miete_m_hh_ab_2009",
+    "ges_rente_zugangsfaktor",
+    "_arbeitsl_geld_2_alleinerz_mehrbedarf_m_hh",
+    "eink_st_altersfreib_ab_2005",
+    "vorsorgeaufw_ab_2020",
+}
```

Comment:
I suspect that these functions cause one problem: Makes sense that the end-to-end test does not run if it uses one of these functions. Regarding the recursion error. I think I used `deepcopy` too often, which somehow caused problems with the `ast.NodeTransformer`. Will be fixed with the next commit.